packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -178,34 +181,44 @@ export function codegenContractFactory(contract: Contract, abi: any, bytecode?:
     connect(signer: Signer): ${contract.name}${FACTORY_POSTFIX} {
       return super.connect(signer) as ${contract.name}${FACTORY_POSTFIX};
     }
-    static connect(address: string, signerOrProvider: Signer | Provider): ${contract.name} {
-      return new Contract(address, _abi, signerOrProvider) as ${contract.name};
-    }
+    static readonly bytecode = "${bytecode.bytecode}";
+    ${body}
   }
 
-  const _abi = ${JSON.stringify(abi, null, 2)};
-
-  const _bytecode = "${bytecode.bytecode}";
-
   ${generateLibraryAddressesInterface(contract, bytecode)}
   `
 }
 
 export function codegenAbstractContractFactory(contract: Contract, abi: any): string {
+  const { body, header } = codegenCommonContractFactory(contract, abi)
   return `
   import { Contract, Signer } from "ethers";
   import { Provider } from "@ethersproject/providers";
-
-  import type { ${contract.name} } from "../${contract.name}";
+  import type { ${contract.name}, ${contract.name}Interface } from "../${contract.name}";
+  ${header}
 
   export class ${contract.name}${FACTORY_POSTFIX} {
-    static connect(address: string, signerOrProvider: Signer | Provider): ${contract.name} {
-      return new Contract(address, _abi, signerOrProvider) as ${contract.name};
-    }
+    ${body}
   }
+  `
+}
+
+function codegenCommonContractFactory(contract: Contract, abi: any): { header: string; body: string } {
+  const header = `
+  import { Interface } from '@ethersproject/abi';
```

Comment:
(1) For concrete contract factories, this import is doubled in the `ethersContractImports` variable
(2) it should be imported `from 'ethers'` (same as the ones in (1)), to avoid problems with ethers subpackages being out of sync, etc